openmdao/core/tests/test_discrete.py

```diff
@@ -718,6 +718,59 @@ def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
         # minimum value
         assert_rel_error(self, prob['SolidityComp.blade_solidity'], 0.02984155, 1e-4)
 
+    def test_feature_discrete_implicit(self):
+
+        import openmdao.api as om
+
+        class ImpWithInitial(om.ImplicitComponent):
+            """
+            An implicit component to solve the quadratic equation: x^2 - 4x + 3
+            (solutions at x=1 and x=3)
+            """
+            def setup(self):
+                self.add_input('a', val=1.)
+                self.add_input('b', val=-4.)
+                self.add_discrete_input('c', val=3)
+                self.add_output('x', val=5.)
+
+                self.declare_partials(of='*', wrt='*')
+
+            def apply_nonlinear(self, inputs, outputs, residuals, discrete_inputs, discrete_outputs):
+                a = inputs['a']
+                b = inputs['b']
+                c = discrete_inputs['c']
+                x = outputs['x']
+                residuals['x'] = a * x ** 2 + b * x + c
+
+            def linearize(self, inputs, outputs, partials, discrete_inputs, discrete_outputs):
+                a = inputs['a']
+                b = inputs['b']
+                x = outputs['x']
+
+                partials['x', 'a'] = x ** 2
+                partials['x', 'b'] = x
+                partials['x', 'x'] = 2 * a * x + b
+
+            def guess_nonlinear(self, inputs, outputs, resids, discrete_inputs, discrete_outputs):
+                # Default initial state of zero for x takes us to x=1 solution.
+                # Here we set it to a value that will take us to the x=3 solution.
+                outputs['x'] = 5
```

Comment:
I think you should remove the `guess` function from this example..  
It's extraneous for this feature and just makes the example more complicated for the reader.

Reply:
We iterated on this, but the guess_nonlinear is necessary so that the user knows that its signature includes the discrete_inputs and discrete_outputs arguments.